### PR TITLE
fix: raises error when a regression is given a batch with no numerics

### PR DESCRIPTION
**Pull request by ipanepen**
This commit raises an error when `non_nan_indices` is empty. There will only be nan values in the target column during regression, because nan values will be converted into labels for other problem types. During regression, all non-numerics in the target column will be turned into nan values and their rows will be dropped. If all the values in the batch are not numeric, all of the rows will be dropped and the feature transformer will fail. This raises an error before that happens.

*Issue #, if available:*

*Description of changes:*

## Merge Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your pull request._

- [x] I have read the [CONTRIBUTING](https://github.com/aws/sagemaker-scikit-learn-extension/blob/master/CONTRIBUTING.md) doc
- [x] I used the commit message format described in [CONTRIBUTING](https://github.com/aws/sagemaker-scikit-learn-extension/blob/master/CONTRIBUTING.md#committing-your-change)
- [x] I have added tests that prove my fix is effective or that my feature works (if appropriate)
- [ ] I have updated any necessary [documentation](https://github.com/aws/sagemaker-scikit-learn-extension/blob/master/README.rst) (if appropriate)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by wiltonwu** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-sklearn-extension-pr
* Commit ID: fa33af3369a43876175b47ce1edb753466afb435
* Result: SUCCEEDED
* [Build Logs](https://cdvka32a8a.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=f69c3019-e133-4cd7-b8b4-2fd89c3e2765%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*

File: src/sagemaker_sklearn_extension/externals/automl_transformer.py [hunk 1]
```diff
@@ -89,6 +89,12 @@ def transform(self, X):
             y = X[:, self.header.target_column_index]
             y_transformed = self.label_transform(y)
             non_nan_indices = np.arange(y_transformed.shape[0])[~np.isnan(y_transformed)]
+            if non_nan_indices.size == 0:
+                raise ValueError(
+                    "All response values in batch are non-numeric. With a Regression problem type only"
+                    " numeric values are used. Verify the target column is set correctly or adjust the"
+                    " problem type and retry."
```
Comment (wiltonwu) [inline src/sagemaker_sklearn_extension/externals/automl_transformer.py:96]: Nit: I'd rephrase this error message to clarify what you're referring to exactly. Maybe clarify the second sentence that the target column should only contain numeric values? Does this mean that we'll need to retrain the model (i.e. start a whole new Autopilot job)?

**Comment by wiltonwu** [comment]
### AWS CodeBuild CI Report

* CodeBuild project: sagemaker-sklearn-extension-pr
* Commit ID: 1bae1881c13cb6b76888c1426d64d53b489a9eb7
* Result: SUCCEEDED
* [Build Logs](https://cdvka32a8a.execute-api.us-west-2.amazonaws.com/Prod/buildlogs?key=d3173246-74b2-4b1b-a92b-e7f9b0e64651%2Fbuild.log) (available for 30 days)

*Powered by [github-codebuild-logs](https://serverlessrepo.aws.amazon.com/applications/arn:aws:serverlessrepo:us-east-1:277187709615:applications~github-codebuild-logs), available on the [AWS Serverless Application Repository](https://aws.amazon.com/serverless/serverlessrepo/)*
